packages/next/src/server/use-cache/use-cache-wrapper.ts

```diff
@@ -694,6 +695,11 @@ export function cache(
         const implicitTags = workUnitStore?.implicitTags
         const forceRevalidate = shouldForceRevalidate(workStore, workUnitStore)
 
+        // Lazily refresh the tags for the cache handler that's associated with
+        // this cache function. This is only done once per request and cache
+        // handler, when it's awaited for the first time.
+        await workStore.refreshTagsByCacheKind.get(kind)
+
```

Comment:
This code is relatively hard to understand because it's not clear that some action is maybe kicked off here without being aware that the value in this map is a lazy promise. Additionally it forces a microtask even when there is no refresh tag support or when the refresh tag has already been awaited during this request. I think we should extract this into it's own helper and do the trick where we stash the settled result so we can synchronously bypass the check on subsequent cache calls

Reply:
addressed in #77843